src/components/Farmhand/Farmhand.js

```diff
@@ -271,19 +270,21 @@ const applyPriceEvents = (valueAdjustments, priceCrashes, priceSurges) => {
  */
 
 export default class Farmhand extends Component {
-  // Bind event handlers
-
-  localforage = localforage.createInstance({
-    name: 'farmhand',
-    description: 'Persisted game data for Farmhand',
-  })
-
   /*!
    * @member farmhand.Farmhand#state
    * @type {farmhand.state}
    */
   state = this.createInitialState()
 
+  static defaultProps = {
```

Comment:
I tend to prefer the `Farmhand.deafultProps` at the end of the file approach just for sake of consistency since it's the only way to set a static property for the functional components. just a thought. if you prefer this way it's ultimately the same

Reply:
My preference is actually the opposite, as I think it's worth leveraging language-level features wherever it seems reasonable (`static` properties, in this case). To me it helps to indicate intent. That being said, I'm fine with a mix of `static` properties and `SomeObject.staticProperty` forms in the code base, so feel free to use it when you need to define a static property!

> it's the only way to set a static property for the functional components

That's a really good point. I actually didn't realize that functional components supported `defaultProps`. Personally, I'd avoid defining `defaultProps` for functional components and instead use default function parameters like we have here: https://github.com/jeremyckahn/farmhand/blob/734284da8b6224f6889bb6612fadb6d999b3004c/src/components/Plot/Plot.js#L77-L81

It solves the same problem, but this form allows the code to read more top-to-bottom.

In any case, I think that both forms work just fine and can be freely mixed. 🙂

Reply:
what you pointed out is fine for defaults, but doesn't work for `propTypes` (which is what i'm now realizing i had in mind when I left the original comment.) either way though this is definitely just a preference choice 😄

Reply:
Ah yes, that's a good point. `propTypes` would have to be defined separately for functional components!